Homework_8/task_3.py

```diff
@@ -0,0 +1,68 @@
+# Решить задачи, которые не успели решить на семинаре.
+# Напишите функцию, которая получает на вход директорию и рекурсивно обходит её и все вложенные директории.
+# Результаты обхода сохраните в файлы json, csv и pickle. Для дочерних объектов указывайте родительскую директорию.
+# Для каждого объекта укажите файл это или директория. Для файлов сохраните его размер в байтах,
+# а для директорий размер файлов в ней с учётом всех вложенных файлов и директорий.
+# Соберите из созданных на уроке и в рамках домашнего задания функций пакет для работы с файлами разных форматов.
+import csv
+import json
+import os
+import pickle
+from pathlib import Path
+
+
+def made_json_csv_pickle(directory, lst):
+    with(
+        open(f'out/{directory.name}_1.json', 'w', encoding='utf8') as json_wf,
+        open(f'out/{directory.name}_2.csv', 'w', encoding='utf8') as csv_wf,
```

Comment:
Не забывайте указывать параметр `newline`, чтобы запись в csv происходила в каждой строке.
```suggestion
        open(f'out/{directory.name}_2.csv', 'w', newline='', encoding='utf8') as csv_wf,
```